Firestore/Source/API/FSTUserDataConverter.mm

```diff
@@ -593,10 +594,28 @@ - (void)parseSentinelFieldValue:(FIRFieldValue *)fieldValue context:(FSTParseCon
           @"SetOptions.merge()%@",
           [context fieldDescription]);
     }
+
   } else if ([fieldValue isKindOfClass:[FSTServerTimestampFieldValue class]]) {
     [context appendToFieldTransformsWithFieldPath:*context.path
                                transformOperation:absl::make_unique<ServerTimestampTransform>(
                                                       ServerTimestampTransform::Get())];
+
+  } else if ([fieldValue isKindOfClass:[FSTArrayUnionFieldValue class]]) {
+    std::vector<FSTFieldValue *> parsedElements =
+        [self parseArrayTransformElements:((FSTArrayUnionFieldValue *)fieldValue).elements];
+    [context appendToFieldTransformsWithFieldPath:*context.path
+                               transformOperation:absl::make_unique<ArrayTransform>(
+                                                      TransformOperation::Type::ArrayUnion,
+                                                      parsedElements)];
```

Comment:
Originally I tried to do have an intermediate variable for the transform operation...

```
std::unique_ptr<TransformOperation> op = absl::make_unique<ArrayTransform>(
                                                      TransformOperation::Type::ArrayUnion,
                                                      parsedElements);
[context appendToFieldTransformsWithFieldPath:*context.path transformOperation:op];
```
But that complained because std::unique_ptr doesn't have a copy constructor.  Does that mean there's no way to create an intermediate variable?  Or is there a trick?